src/librustc/ty/mod.rs

```diff
@@ -1535,6 +1541,7 @@ where
     Types(I),
     InputTypes(J),
     ProjectionTypes(K),
+    WellFormedConst(Ty<'tcx>, I),
```

Comment:
`Predicate::walk_tys` is even more cursed than `Ty::walk`, jeez. At least I think your fix is okay for now.

Reply:
FWIW, #70164 removes `walk_tys` completely, it was a rustdoc-only fragile hack.